extensions/security/runtime/src/main/java/io/quarkus/security/runtime/SecurityConfig.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.security.runtime;
+
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+/**
+ * Quarkus Security configuration.
+ */
+@ConfigMapping(prefix = "quarkus.security")
+@ConfigRoot(phase = ConfigPhase.RUN_TIME)
+public interface SecurityConfig {
+
+    /**
+     * Security events configuration.
+     */
+    SecurityEventsConfig events();
+
+    interface SecurityEventsConfig {
+
+        /**
+         * Whether security events should be fired.
+         */
+        @WithDefault("true")
+        boolean enabled();
```

Comment:
Do we really need this property ? Users express that it is required indirectly by registering observers.

Reply:
> Do we really need this property ? 

There is very small cost to having this property, but there is nice advantage - you can only enable debugging, which can be costly, without any code changes, you can do it at runtime for example on just some containers etc.

> Users express that it is required indirectly by registering observers.

They express they want to consume events in certain situations, what if they don't need it always?

Reply:
> @michalvavrik Right, I agree, users sometimes may want to disable some security features for example, such as authorization, or, when testing, they use @TestSecurity or Wiremocks

This is not only about disabling security features, there is expense of firing events because for example we need to catch exceptions and create event instances etc. You only want to do this optionally when you need it. Like if you experience some bug in a service, so you sent environment variable that enables security events.

> but I'm not sure how this property helps, theoretically at least, they can have Observes but for some reasons don't want anything reported, even if the security capability is present - if it is not they won't be reported - which is why I'm not sure it is necessary to introduce it immediately to address what looks like a rather rare case (Security capability is present but users don't want events fired).

How is this different to optionally enabling debug or trace logging? It is more effective to allow users to disable security events instead of for example define observers that log everything with debug / trace because you avoid some processing.

> May be it should be dropped for now - but it can easily be re-introduced if someone ever asks about such an option ?

Well, I'll do it if you insist. I think we are cutting down cool feature which users can't implement themselves. If they introduce application-specific custom config property that inside observer method skip processing of events when they don't need it, it is after we have already caught exception and collected data like `Securityidentity` and set the event.

If you think users can show their interest at build time (by defining observers), I'm not sure why there are runtime properties disabling features like `quarkus.hibernate-orm.active` and others.

Reply:
@michalvavrik 

>I think we are cutting down cool feature

I can guarantee you I'm not on a mission to cut down the coolness from this PR :-) I'm just looking what can be dropped which may not necessarily needed if needed at all, I know I'm not always following it myself :-).

My understanding properties like `quarkus.hibernate-orm.active=false` or `quarkus.oidc.tenant-enabled=false` etc are about making sure the application can start where no given DB is available or no OIDC provider is listening, etc. 

I'm just finding it hard to imagine why users would want to avoid observer methods firing at runtime. Is it because the downstream channel these event observers are coded to report to, ex kafka, that is expecting them is not available in this case, something like that ? May be then it is indeed necessary

Reply:
@michalvavrik To be honest I don't find these 3 cases worthy of having a dedicated configuration property since IMHO users have good reasons for observing events in the code beyond typing it all in the prod code just to keep an option to debug. But IMHO I'd rather recommend users disable events in cases where their consumers are not available rather than injecting configuration properties - this is a case which I'm thinking about which might justify this property IMHO

Reply:
> To be honest I don't find these 3 cases worthy of having a dedicated configuration property since IMHO users have good reasons for observing events in the code beyond typing it all in the prod code just to keep an option to debug.

Alright, I hear you and disagree. However I can accept dropping this property if you insist, please let me know.

Reply:
We have conclusion, thanks.